activemodel/lib/active_model/validations/numericality.rb

```diff
@@ -68,18 +69,24 @@ def validate_each(record, attr_name, value)
     private
 
       def is_number?(raw_value)
-        !parse_raw_value_as_a_number(raw_value).nil?
+        !parse_as_number(raw_value).nil?
```

Comment:
That does read better, but I think because the `raw_value` can be an _anything_ (including another ActiveRecord object by mistake), its best not to use `#present?` for performance reasons.